lib-injection/README.md

```diff
@@ -0,0 +1,30 @@
+# Host injection
+
+Our gitlab build pipeline ships a pre-installed tracer deb/rpm package.
+
+Currently, we support
+
+| Environment| version |
+|---|---|
+| Ruby  | `2.7`, `3.0`, `3.1`, `3.2`|
+| OS    | Debian 10+ |
+| Arch  | `amd64` |
+
+
+### Packaging
+
+In order to ship `ddtrace` and its dependencies as a pre-install package, we need a few tweaks in our build pipeline.
+
+* Use multiple custom built Ruby images to build native extensions. Those images are based on Debian `buster` to support older distribution and Ruby is compiled as a static library with `--disbale-shared` option which disables the creation of shared libraries (also known as dynamic libraries or DLLs).
+* Install `ffi` gem with its built-in `libffi` instead of system `libffi`.
+* After installation, rename directories to remove the `-static` suffix so user’s ruby can detect those  `.so` files and make sure files have read permission.
```

Comment:
Who creates `-static` directories?
Is it a specific part of `ddtrace`?
If so, can we add it here?

Reply:
When `gem install msgpack` with Ruby compiled with `--disabled-shared`, the native extension would be stored at `extensions/x86_64-linux/3.2.0-static/` instead of `extensions/x86_64-linux/3.2.0/`. 

This would be an issue for other Ruby installation to lookup for native extension. Eventually, it would raise error about could not find native extension but gem has been installed.